packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -177,14 +178,30 @@ function optimiseIndexGetter(path, argsId, offset) {
     scope.push({ id: temp, kind: "var" });
     path.parentPath.replaceWith(restIndexImpure({
       ARGUMENTS: argsId,
+      OFFSET: offsetLiteral,
       INDEX: index,
       REF: temp,
     }));
   } else {
-    path.parentPath.replaceWith(restIndex({
+    const parentPath = path.parentPath;
+    parentPath.replaceWith(restIndex({
       ARGUMENTS: argsId,
+      OFFSET: offsetLiteral,
       INDEX: index,
     }));
+
+    // See if we can statically evaluate the first test (i.e. index < offset)
+    // and optimize the AST accordingly.
+    const offsetTestPath = parentPath.get("test").get("left");
+    const valRes = offsetTestPath.evaluate();
+    if (valRes.confident) {
+      if (valRes.value === true) {
+        parentPath.replaceWith(parentPath.scope.buildUndefinedNode());
+      }
+      else {
+        parentPath.get("test").replaceWith(parentPath.get("test").get("right"));
```

Comment:
See this https://github.com/babel/babel/pull/5810/files#diff-171b2ff6e4556995af8a3b21cf2209dcR5.

If you comment out `parentPath.get("test").replaceWith(parentPath.get("test").get("right"));`, the generated code would be: `1 < 1 || arguments.length <= 1 ? undefined : arguments[1];`.